tests/handlers/test_presence.py

```diff
@@ -716,19 +718,41 @@ def test_set_presence_from_syncing_keeps_status(self):
         # our status message should be the same as it was before
         self.assertEqual(state.status_msg, status_msg)
 
-    def test_set_presence_from_syncing_keeps_busy(self):
+    @parameterized.expand([(False,), (True,)])
+    @unittest.override_config(
+        {
+            "experimental_features": {
+                "msc3026_enabled": True,
+            },
+        }
+    )
+    def test_set_presence_from_syncing_keeps_busy(self, test_with_workers: bool):
         """Test that presence set by syncing doesn't affect busy status"""
-        # while this isn't the default
-        self.presence_handler._busy_presence_enabled = True
-
         user_id = "@test:server"
         status_msg = "I'm busy!"
 
+        worker_hs = None
+        if test_with_workers:
+            # Create a worker that will handle /sync traffic
+            worker_hs = self.make_worker_hs(
+                "synapse.app.generic_worker", {"worker_name": "presence_writer"}
+            )
+
         self._set_presencestate_with_status_msg(user_id, PresenceState.BUSY, status_msg)
 
-        self.get_success(
-            self.presence_handler.user_syncing(user_id, True, PresenceState.ONLINE)
-        )
+        if test_with_workers and worker_hs is not None:
```

Comment:
This can be simplified since `worker_hs` is only not `None` if `test_with_workers` is true:

```suggestion
        if worker_hs is not None:
```

Reply:
Done in 75089ded5d7f2461387842226c5088ecf3f667f0.